### PR TITLE
CMakeLists: Use boost 1.79.0 instead

### PR DESCRIPTION
**Pull request by Morph1984**
Initial testing has uncovered that boost 1.81.0 may have been the source of some issues, reverting this for now.

https://github.com/yuzu-emu/build-environments/pull/71

### CODE REVIEW

**Review by liamwhite — APPROVED** [review]

**Comment by liamwhite** [comment]
Note for distro packagers: we need some time to figure out what the issue was, until now this unbreaks our CI

**Comment by Grimish-ng** [comment]
Getthing the same on my arch build-- Target architecture: x86_64
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost: Found unsuitable version "1.81.0", but required is
  exact version "1.79.0" (found /usr/include, found components: context)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:598 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindBoost.cmake:2377 (find_package_handle_standard_args)
  CMakeLists.txt:213 (find_package)


-- Configuring incomplete, errors occurred!
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'yuzu-mainline-git-r23453.76289c2e9-1': 
error: packages failed to build: yuzu-mainline-git-r23453.76289c2e9-1

This is from the arch repo package boost I believe.

**Comment by liamwhite** [comment]
Boost 1.81 is severely broken. We are investigating what the issue is.

**Comment by abouvier** [comment]
Broken how? I compiled yuzu 1382 with boost 1.81 and it's working fine.

File: CMakeLists.txt [hunk 1]
```diff
@@ -210,7 +210,7 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 # =======================================================================
 
 # Enforce the search mode of non-required packages for better and shorter failure messages
-find_package(Boost 1.81.0 REQUIRED context)
+find_package(Boost 1.79.0 EXACT REQUIRED COMPONENTS context)
```
Comment (abouvier) [inline CMakeLists.txt:213]: `COMPONENTS` is not needed when `REQUIRED` is present ;)

**Comment by Morph1984** [comment]
> Broken how? I compiled yuzu 1382 with boost 1.81 and it's working fine.

See this https://github.com/boostorg/boost/issues/757

It took awhile to figure this out

**Comment by Morph1984** [comment]
We are no longer requiring exactly 1.79.0 now. However, we will require a minimum of 1.79.0 and our buildbots will remain on this version until an upgrade to other libraries is mandatory.

**Comment by Morph1984** [comment]
Also, huge thanks to @liamwhite for bisecting the issue, narrowing it down to the container_hash changes in boost 1.81.0

**Comment by Kelebek1** [comment]
Now that the cause is known and w ehave the old hashing, can you leave us on 1.81.0?

**Comment by noloader** [comment]
Hi Everyone.

I find myself in a Boost quandary, too. I want to build the latest Yuzu from sources to perform some testing. I want to use some custom CFLAGS and CXXFLAGS to carry out the tests.

Ubuntu 22.04 (LTS) only provides Boost 1.74. Next, I tried Fedora 37. Fedora 37 only supplies Boost 1.78. Ubuntu 22.04 in not even a year old, and Fedora 37 is not even 3 months old. Neither can meet the requirements laid down by Yuzu.

I'm guessing Linux from Scratch, Gentoo and some other distros may be able to meet Yuzu's requirements. But I am not into building my own platform. It takes me two or three days to get a Gentoo box running. I think it is a lot to ask of someone just to test Yuzu.

To support contemporary Ubuntu and Fedora, Boost 1.71 would be a good choice. Fedora is easy because its release cadence is every 6 months. Ubuntu is a little trickier because of its LTS support. But a Boost 1.71 minimum would satisfy both Ubuntu 20 and Fedora 36.

So the ask is, please consider what is available on contemporary distros when setting requirements, like Boost 1.81.

-----

Sorry for tacking the Boost version comment here. I did not want to open an issue to bring this to the team's attention.

**Pull request merged**